.github/workflows/e2e.yml

```diff
@@ -244,3 +244,96 @@ jobs:
             }
         env:
           SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+  run-generator-private-client:
+    strategy:
+      fail-fast: false
+      # Run one matrix env at a time because we need to deploy each app to MRT and run e2e tests there
+      max-parallel: 1
```

Comment:
Could you talk me through this point a bit more? I am not sure why we have to run each job one at a time.

I'd assume this would take a really long time to do since we're running the tests over a matrix of node and npm one at a time?

Will this slow down the CI for PRs or will this just run for nightlies?

Reply:
So we have 4 jobs:
1. Run generator using public client preset (no mrt deploy)
1. Run generator public client using generator responses NO extensibility (deploys to MRT -> playwright tests run against the deployment)
1. Run generator public client using generator responses WITH extensibility (deploys to MRT -> playwright tests run against the deployment)
1. Run generator using private client preset (deploys to MRT -> playwright tests run against the deployment)

Now,
Each of these by themselves are run about 8 times for combinations of node matrix
1, 2, 4 run in parallel as overall jobs while 3 needs to wait for 2 to finish since they both deploy to same MRT instances.
Now diving into each of these:
2, 3 & 4 deploy to mrt
So 2 generates app on node 18, npm 9 (this is one of 8 matrix combinations)
Deploys app to MRT
Runs playwright test
Switches Node versions to next matrix combinations
Repeat

See how each env within each job generates an app and deploys to MRT. All these deploys for public clients happen on the same MRT instance.
So max-parallel=1 means, within the job, the matrix executions must happen one at a time.